sql/rowexec/ddl_iters.go

```diff
@@ -1859,6 +1859,27 @@ func (b *BaseBuilder) executeDropCheck(ctx *sql.Context, n *plan.DropCheck) erro
 		return err
 	}
 
+	checkTable, ok := chAlterable.(sql.CheckTable)
+	if !ok {
+		return plan.ErrNoCheckConstraintSupport.New(chAlterable.Name())
+	}
+
+	checks, err := checkTable.GetChecks(ctx)
+	if err != nil {
+		return err
+	}
+
+	exists := false
+	for _, check := range checks {
+		if strings.EqualFold(check.Name, n.Name) {
+			exists = true
+		}
+	}
+
+	if !exists && n.IfExists {
```

Comment:
Seems strange to do the validation logic here but only for the IfExists case. Should also throw an error if the check isn't found and that flag isn't set.

Reply:
Yeah, right now it's relying on `chAlterable.DropCheck` to return the does not exist error. You're right that we've done the work already though and could return that error without having to recheck. I'll add that error here.